src/custom-element.js

```diff
@@ -53,6 +53,13 @@ const PREPARE_LOADING_THRESHOLD_ = 1000;
 
 
 /**
+ * List of tag names of elements that are not allowed to be AMP placeholders.
+ * @private @const {!RegExp}
+ */
+const BLACKLISTED_PLACEHOLDER_TAGS_REGEX_ = /^(?:input|select|textarea)$/i;
```

Comment:
Went with @sriramkrish85 suggestion of just checking if the element natively have `placeholder` property. So dropped the blacklist.